mkosi/config.py

```diff
@@ -1575,20 +1590,8 @@ def parse_config(self, path: Path, namespace: argparse.Namespace) -> bool:
                         if not (match := s.match):
                             die(f"{k} cannot be used in [Match]")
 
-                        # If we encounter a setting in [Match] that has not been explicitly configured yet,
-                        # we assign the default value first so that we can [Match] on default values for
-                        # settings.
-                        if s.dest not in namespace:
-                            if s.default_factory:
-                                default = s.default_factory(namespace)
-                            elif s.default is None:
-                                default = s.parse(None, None)
-                            else:
-                                default = s.default
```

Comment:
Basically you should add another if condition here (that takes precedence over all these) which uses the value from the defaults namespace if it is in the defaults namespace.